### DISCUSSION TITLE
Is it possible to model the problem of impact of a crushing body against a rigid surface?

### DISCUSSION THREAD

**Discussion by Dadidal** [Anura3D/Anura3D_OpenSource#172] [General]
Hello! I am a PhD student and my thesis aims to model the dynamic problem of the impact of crushing a deformable bodies against a rigid surface. My case is that at the beginning of the simulation, I have two bodies (3D model) - deformable solid and rigid surface. During the simulation, the deformable body moves to the rigid surface and crushes onto small parts during the contact. These small parts continue to interact with rigid surface and produce the impact load. Therefore it is assumed that the initial internal nodes of the solid can interact with rigid surface after fracture.
I read a Manual of Anura 3D but I still cannot understand if could I use Anura 3D to simulate the case described above. Could you clarify if is it possible to define such contact in Anura to correctly simulate the crushing event?
Thank you.
![image](https://github.com/Anura3D/Anura3D_OpenSource/assets/32517441/ec67edcd-54bb-4c54-81f7-b5941930453a)

**Comment by ayerrocolom** [comment]
Hello,
Anura3D can deal with the problem of multi-body interaction and large deformations of continuum material. However, the "material crushing" mechanism that you are describing here can be challenging to capture accurately in a continuum framework like MPM depending on the scale of the problem. If the MPM framework is used, then the results will highly depend on the constitutive model considered in the analysis. Note that only simple consecutive models (e.g., linear elasticity, Mohr-Coulomb) are provided in Anura3D, which might not be good enough for your problem (this is something that the user should evaluate). However, the user can definitely incorporate a differently defined constitutive model if desired.

**Comment by Dadidal** [comment]
@ayerrocolom Thank you for your answer!
I understand about the constitutive model, it is one of the planned features that I want to introduce to my model.

> Anura3D can deal with the problem of multi-body interaction and large deformations of continuum material

Could you please clarify if can Anura3D deal with multiple fractures and the separation of solid bodies into small bodies, which then will interact with each other?
I understand about the multi-body interaction possibilieties of Anura3D when the bodies are predefined before calculation. But I want to model the case when I have one initial body, which will be fractured and separated.
